### PR TITLE
Featurebranch

### PR DESCRIPTION
**Pull request by BirSikanderMahajan**
Updated the code

--if(jsonSchedule.get("Type").getAsString().contentEquals("TemporaryScheduleInfo")) continue;

### CODE REVIEW

File: src/main/java/com/Leapwork/Leapwork_plugin/Messages.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    public static final String TIME_DELAY_NUMBER_IS_INVALID = "Time delay number is invalid: %1$s, setting to default %2$s";
+    public static final String FULL_REPORT_FILE_PATH = "Creating report file by path: %1$s";
+
+    public static final String SCHEDULE_DISABLED = "Schedule %1$s[%2$s] is disabled!";
+
+    public static final String INVALID_SCHEDULE_VARIABLE = "Failed to parse variable: %1$s";
+    public static final String SCHEDULE_VARIABLE_KEY_DUPLICATE = "Key value pair with the same key already exists: %1$s. This pair will be ignored";
+    public static final String SCHEDULE_VARIABLE_REQUEST_PART = "Schedule variables request part: %1$s";
+
+    public static final String INPUT_VALUES_MESSAGE = "LeapWork Plugin input parameters:";
```
Comment (AntonLiakhovich) [inline src/main/java/com/Leapwork/Leapwork_plugin/Messages.java:89]: LeapWork or Leapwork?

File: src/main/java/com/Leapwork/Leapwork_plugin/Messages.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    public static final String INPUT_HOSTNAME_VALUE = "LEAPWORK controller hostname: %1$s";
```
Comment (AntonLiakhovich) [inline src/main/java/com/Leapwork/Leapwork_plugin/Messages.java:90]: Change to Leapwork everywhere

File: README.md [hunk 3]
```diff
@@ -1,2 +1,79 @@
...
+ - Generate a xml report file in JUnit format
+ - Write tests trace to build output log
+ - Smart UI
+ 
+# Installing
+- Use maven 
+- Command: mvn package 
+- Or simply install hpi-file from the "target" folder: Manage Jenkins -> Manage Plugins -> Advanced -> Upload Plugin -> Choose that hpi-file -> Press Upload
+
+# Update 4.0.1
```
Comment (AntonLiakhovich) [inline README.md:21]: In release notes, keep only:
- Added support for Leapwork version 2021.1 and newer
- Plugin now uses Leapwork API v4

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,160 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+	<modelVersion>4.0.0</modelVersion>
```
Comment (AntonLiakhovich) [inline pom.xml:5]: so _4.0.0_ or _4.0.1_?

File: src/main/java/com/Leapwork/Leapwork_plugin/Messages.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
+package com.Leapwork.Leapwork_plugin;
+
+public class Messages {
+
+    public static final String SCHEDULE_FORMAT = "%1$s[%2$s]";
+    public static final String SCHEDULE_DETECTED = "Schedule %1$s[%2$s] successfully detected!";
+    public static final String SCHEDULE_RUN_SUCCESS = "Schedule %1$s[%2$s] Launched Successfully!";
```
Comment (AntonLiakhovich) [inline src/main/java/com/Leapwork/Leapwork_plugin/Messages.java:7]: Launched  -> riggered

File: src/main/java/com/Leapwork/Leapwork_plugin/Messages.java [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+    public static final String SCHEDULE_TITLE_OR_ID_ARE_NOT_GOT = "Tried to get schedule title or id!";
```
Comment (AntonLiakhovich) [inline src/main/java/com/Leapwork/Leapwork_plugin/Messages.java:9]: Remove exclamation mark from messages

File: src/main/java/com/Leapwork/Leapwork_plugin/Messages.java [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+    public static final String STOP_SCHEDULE_FAIL = "Failed to stop schedule %1$s[%2$s]!";
+
+    public static final String STOP_RUN_SUCCESS = "Schedule %1$s run %2$s stopped successfully!";
+
+    public static final String STOP_RUN_FAIL = "Failed to stop schedule %1$s run %2$s!";
+
+    public static final String INVALID_ACCESS_KEY = "Invalid or empty access key!";
+
+    public static final String DATABASE_NOT_RESPONDING = "Data base is not responding!";
```
Comment (AntonLiakhovich) [inline src/main/java/com/Leapwork/Leapwork_plugin/Messages.java:72]: _Data base_ should in one word

File: target/classes/index.jelly [hunk 8]
```diff
@@ -0,0 +1,5 @@
+<?jelly escape-by-default='true'?>
```
Comment (AntonLiakhovich) [inline target/classes/index.jelly:1]: why do we need all these file (this file and all others) in _target_ folder? how is plugin installing? does it need all these files or build output is one file?

File: README.md [hunk 9]
```diff
@@ -1,2 +1,79 @@
...
+ - Automatically receive test results
+ - Build status based tests results
+ - Generate a xml report file in JUnit format
+ - Write tests trace to build output log
+ - Smart UI
+ 
+# Installing
+- Use maven 
+- Command: mvn package 
+- Or simply install hpi-file from the "target" folder: Manage Jenkins -> Manage Plugins -> Advanced -> Upload Plugin -> Choose that hpi-file -> Press Upload
```
Comment (AntonLiakhovich) [inline README.md:19]: where is this file in _target_ folder?

**Review by AntonLiakhovich — CHANGES_REQUESTED** [review]
Check all my comments

File: src/main/java/com/Leapwork/Leapwork_plugin/Messages.java (hunk 1 above)
Comment (BirSikanderMahajan) [reply src/main/java/com/Leapwork/Leapwork_plugin/Messages.java:89]: Updated in branch : updatefeaturebranch

File: README.md (hunk 3 above)
Comment (BirSikanderMahajan) [reply README.md:21]: Updated in branch : updatefeaturebranch

File: pom.xml (hunk 4 above)
Comment (BirSikanderMahajan) [reply pom.xml:5]: This 4.0.0 was added from the template that was used to create the plugin

File: src/main/java/com/Leapwork/Leapwork_plugin/Messages.java (hunk 5 above)
Comment (BirSikanderMahajan) [reply src/main/java/com/Leapwork/Leapwork_plugin/Messages.java:7]: Updated in branch : updatefeaturebranch

File: src/main/java/com/Leapwork/Leapwork_plugin/Messages.java (hunk 6 above)
Comment (BirSikanderMahajan) [reply src/main/java/com/Leapwork/Leapwork_plugin/Messages.java:9]: Updated in branch : updatefeaturebranch

File: src/main/java/com/Leapwork/Leapwork_plugin/Messages.java (hunk 7 above)
Comment (BirSikanderMahajan) [reply src/main/java/com/Leapwork/Leapwork_plugin/Messages.java:72]: Updated in branch : updatefeaturebranch

File: target/classes/index.jelly (hunk 8 above)
Comment (BirSikanderMahajan) [reply target/classes/index.jelly:1]: this jelly file is the main file, the text written here is visible text when we update the hpi file

File: README.md (hunk 9 above)
Comment (BirSikanderMahajan) [reply README.md:19]: Updated in branch : updatefeaturebranch

File: target/classes/com/Leapwork/Leapwork_plugin/LeapworkJenkinsBridgeBuilder/help-leapworkHostname.html [hunk 10]
```diff
@@ -0,0 +1,3 @@
+<div>
+Enter Leapwork Controller hostname or IP !
```
Comment (AntonLiakhovich) [inline target/classes/com/Leapwork/Leapwork_plugin/LeapworkJenkinsBridgeBuilder/help-leapworkHostname.html:2]: there is still an exclamation mark

**Review by AntonLiakhovich — COMMENTED** [review]
one last bit

**Comment by AntonLiakhovich** [comment]
Approved

**Pull request merged**